pkg/controller/common/helpers.go

```diff
@@ -150,13 +129,47 @@ func WriteTerminationError(err error) {
 	glog.Fatal(msg)
 }
 
-// convertIgnition3to2 takes an igntion v3 config and returns a v2 config
-func convertIgnition3to2(ignconfig ign3types.Config) (ign2types.Config, error) {
-	converted2, err := ignconverter.Translate3to2(ignconfig)
+// ConvertIgnition2to3 takes an ignition spec v2 config and returns a v3 config
+func ConvertIgnition2to3(ign2config ign2types.Config) (ign3types.Config, error) {
+	// only support writing to root file system
+	fsMap := map[string]string{
+		"root": "/",
+	}
+
+	// Workaround to get v2.3 as input for converter
+	ign2_3config := ign2_3.Translate(ign2config)
+	ign3_0config, err := ignconverter.Translate(ign2_3config, fsMap)
+	if err != nil {
+		return ign3types.Config{}, errors.Errorf("unable to convert Ignition spec v2 config to v3: %v", err)
+	}
+	// Workaround to get a v3.1 config
+	converted3 := translate3.Translate(ign3_0config)
+
+	glog.V(4).Infof("Successfully translated Ignition spec v2 config to Ignition spec v3 config: %v", converted3)
+	return converted3, nil
+}
+
+// ConvertIgnition3to2 takes an ignition spec v3 config and returns a v2 config
+func ConvertIgnition3to2(ign3config ign3types.Config) (ign2types.Config, error) {
+	// Bad hack to convert spec 3.1 config to spec 3.0
+	// ign-convert doesn't currently support spec v3.1
```

Comment:
why do we even need to support 3.1?

Reply:
I think support 3.1 itself isn't a hack, just that 3.0 > 3.1 doesn't translate very nicely. I suppose while we're developing this it'd be good to keep to latest stable

Reply:
Would make sense to just support latest stable instead of translating down to it (from a messaging perspective & from a support perspective).

Reply:
Right we are supporting 3.1 directly via this change. The hack in place is for internal purposes to support 3.0 (which is used for the translator)

Reply:
The main reason to go directly to v3.1 is that installer uses (or will use rather) some 3.1 functionality (it currently shims to v2.4 on some platforms for lack of those features in v2.2). It would be nice if we could get rid of that shim there and agree on spec v3.1 being used universally. (v3.1 also has vastly simplified and improved typing).

It would be cleanest here to wait for https://github.com/coreos/ign-converter/pull/6 and use that to avoid this hack here.

Reply:
Ahh gotcha. So if we really want 3.1 and are just waiting for that PR I think we could wait a little bit. Seems like it's basically finished and just needs a final review? If it gets stuck we can stop waiting ofc but seems really close to merging?

Reply:
(also didnt realize 3.1 _was_ stable)

Reply:
updated ign-converter and removed the hack

Reply:
yay!